bottomless/src/replicator.rs

```diff
@@ -317,27 +321,31 @@ impl Replicator {
             })
         };
 
+        let (upload_progress, last_uploaded_frame_no) = CompletionProgress::new(0);
```

Comment:
why is baseline necessarily 0? could we be in the middle of a generation?

Reply:
When we initialise it, it's always 0. If we're in a middle of the generation, we know about this as part of restore procedure, which happens after initialisation. Then change below checks out the local status of a frame counts and updates it: 
```rust
{
    let mut guard = self.upload_progress.lock().await;
    guard.update(1, last_frame);
}
```